corgi/core/models.py

```diff
@@ -1696,13 +1696,16 @@ def get_provides_nodes_queryset(
         )
         if include_dev:
             type_list = ComponentNode.PROVIDES_NODE_TYPES
-        return (
-            self.cnodes.get_queryset()
-            .get_descendants()
-            .filter(type__in=type_list)
-            .prefetch_related("obj")
-            .using(using)
-        )
+
+        provides_set = set()
+        for cnode in self.cnodes.get_queryset():
+            provides_set.update(
+                cnode.get_descendants()
+                .filter(type__in=type_list)
+                .using(using)
+                .values_list("pk", flat=True)
+            )
```

Comment:
Let's just change it to:

provides_set = self.get_provides_nodes()

Reply:
ok I did not know if you were going to rationalise into a single func ... will do